Nullpointer exception when parsing endpoint for a local instance of DynamoDB

F2006 opened the pull request:
When connecting to a local instance of DynamoDB, the AwsHostNameUtil.parseRegion returns null, since the endpoint is not a valid AWS endpoint. This null is then passed down to Regions.fromName and then an NPE is thrown and thus preventing the DynamoDBDelegate class from being instantiated.

amcp commented:
Looks good - please squash all commits to one. And add your name to the contributors list in pom.xml. I'll review once more and then merge in!

F2006 commented:
I have squashed the commits and updated the contributor list.

I had a look at the build failing. Now that a connection is actually opened to Dynamo locally, it seems like the local instance of Dynamo is not there anymore when the tests run. Even though from the logs, it was started before the tests. I ran the tests locally a few times and it could connect fine. Will need to investigate what is happening with the build on Travis.

There seems to be permission issues as well further down in the logs:
Caused by: org.elasticsearch.ElasticsearchIllegalStateException: Failed to obtain node lock, is the following location writable?: [/tmp/searchindex/data/elasticsearch]

If there is anything I can help with to get the build going again, let me know.

amcp commented:
@F2006 please create an issue for using a different directory for the ES root as it makes sense that I don't have permission to write to /tmp in the Travis container.

amcp commented:
@F2006 You could fork the project (you already did) and set up your own Travis CI for the time being. But I can work on the ES permission issue to start. Good catch.

amcp commented:
@F2006 created #80 to track the /tmp/elasticsearch fix
